File: apps/Card/shared/CardCore/Sources/CardCore/Observability/AnalyticsService.swift
// AnalyticsService — canImport-gated stub. When PostHog is present at compile
// time, the wrapper forwards events; otherwise it is a no-op so CardCore
// builds without any third-party deps. See SENTRY.md to actually wire PostHog.
import Foundation

/// The capture surface that triggered the event. Distinguishing per-surface
/// latency / conversion is the only reason this enum exists.
public enum Surface: String, Sendable {
    case app
    case shareExtension = "share_extension"
    case watch
    case complication
    case tile
}

public enum AnalyticsEvent: Sendable {
    case cardCaptured(surface: Surface, kind: CardKind)
    case cardConverted(from: CardKind, to: CardKind)
    case reminderScheduled(surface: Surface, delayMinutes: Int)
    case reminderFired(surface: Surface)
    case cardDeleted(kind: CardKind)
    case settingsToggled(name: String, enabled: Bool)
    case onboardingCompleted

    public var name: String {
        switch self {
        case .cardCaptured:        return "card_captured"
        case .cardConverted:       return "card_converted"
        case .reminderScheduled:   return "reminder_scheduled"
        case .reminderFired:       return "reminder_fired"
        case .cardDeleted:         return "card_deleted"
        case .settingsToggled:     return "settings_toggled"
        case .onboardingCompleted: return "onboarding_completed"
        }
    }

    public var properties: [String: String] {
        switch self {
        case .cardCaptured(let s, let k):
            return ["surface": s.rawValue, "kind": k.rawValue]
        case .cardConverted(let f, let t):
            return ["from_kind": f.rawValue, "to_kind": t.rawValue]
        case .reminderScheduled(let s, let m):
            return ["surface": s.rawValue, "delay_minutes": String(m)]
        case .reminderFired(let s):
            return ["surface": s.rawValue]
        case .cardDeleted(let k):
            return ["kind": k.rawValue]
        case .settingsToggled(let n, let e):
            return ["name": n, "enabled": String(e)]
        case .onboardingCompleted:
            return [:]
        }
    }
}

public protocol AnalyticsTransport: AnyObject {
    func track(name: String, properties: [String: String])
}

public final class AnalyticsService {
    public static let shared = AnalyticsService()
    public var optedIn: Bool = false
    private var transport: AnalyticsTransport?

    public init() {}

    public func attach(transport: AnalyticsTransport?) { self.transport = transport }

    public func track(_ event: AnalyticsEvent) {
        guard optedIn, let t = transport else { return }
        t.track(name: event.name, properties: event.properties)
    }
}

#if canImport(PostHog)
import PostHog
extension AnalyticsService {
    public func usePostHog(apiKey: String, host: String = "https://us.i.posthog.com") {
        let cfg = PostHogConfig(apiKey: apiKey, host: host)
        PostHogSDK.shared.setup(cfg)
        attach(transport: PostHogTransport())
    }
}
private final class PostHogTransport: AnalyticsTransport {
    func track(name: String, properties: [String: String]) {
        PostHogSDK.shared.capture(name, properties: properties as [String: Any])
    }
}
#endif

File: apps/Card/android/core/src/main/java/com/americangroupllc/card/core/obs/AnalyticsService.kt
package com.americangroupllc.card.core.obs

import com.americangroupllc.card.core.models.CardKind

/** The capture surface that triggered an event. Mirrors Swift's `Surface` enum. */
enum class Surface(val raw: String) {
    APP("app"),
    SHARE_EXTENSION("share_extension"),
    WATCH("watch"),
    COMPLICATION("complication"),
    TILE("tile"),
}

sealed class AnalyticsEvent(val name: String, val properties: Map<String, String>) {
    class CardCaptured(surface: Surface, kind: CardKind) :
        AnalyticsEvent("card_captured", mapOf("surface" to surface.raw, "kind" to kind.name.lowercase()))

    class CardConverted(from: CardKind, to: CardKind) :
        AnalyticsEvent("card_converted", mapOf("from_kind" to from.name.lowercase(), "to_kind" to to.name.lowercase()))

    class ReminderScheduled(surface: Surface, delayMinutes: Int) :
        AnalyticsEvent("reminder_scheduled", mapOf("surface" to surface.raw, "delay_minutes" to delayMinutes.toString()))

    class ReminderFired(surface: Surface) :
        AnalyticsEvent("reminder_fired", mapOf("surface" to surface.raw))

    class CardDeleted(kind: CardKind) :
        AnalyticsEvent("card_deleted", mapOf("kind" to kind.name.lowercase()))

    class SettingsToggled(propName: String, enabled: Boolean) :
        AnalyticsEvent("settings_toggled", mapOf("name" to propName, "enabled" to enabled.toString()))

    object OnboardingCompleted : AnalyticsEvent("onboarding_completed", emptyMap())
}

interface AnalyticsTransport {
    fun track(name: String, properties: Map<String, String>)
}

class AnalyticsService {
    var optedIn: Boolean = false
    private var transport: AnalyticsTransport? = null

    fun attach(t: AnalyticsTransport?) { transport = t }

    fun track(event: AnalyticsEvent) {
        if (!optedIn) return
        transport?.track(event.name, event.properties)
    }

    companion object { val shared = AnalyticsService() }
}
